### DISCUSSION TITLE
Killing a websocket flow

### DISCUSSION THREAD

**Discussion by plongstaff** [mitmproxy/mitmproxy#6375] [General]
I want to kill a websocket flow if a certain message is received. I have an addon which uses the `websocket_message()` hook. In that hook, if I call `flow.kill()`, I get an exception "flow is not killable". Digging into `killable`, I see that `live` is False. 

How should I kill this flow? I would like to terminate the underlying tcp connections.

Thanks,
Phil

**Comment by Prinzhorn** [comment]
I _think_ this is a duplicate of https://github.com/mitmproxy/mitmproxy/issues/4711, the API does not currently allow that.

**Reply by mhils** [reply]
Yes, this is the correct answer. 👍
